magento/pwa-studio
# [PWA-1099] Add support for Product URL and Category URL Settings

## Description

As a admin user, I would like the ability to change settings in the admin panel to control the product URL and category URL, and view it in my Venia storefront.

**Admin Config Settings:**

- Stores > Configuration > Catalog > Catalog > Search Engine Optimization > Product URL Suffix
- Stores > Configuration > Catalog > Catalog > Search Engine Optimization > Category URL Suffix

## Related Issue





https://jira.corp.magento.com/browse/PWA-1099

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Switch store view after directly accessing Category via Home Page, Search Suggestions, Bread Crumbs, Left Nav Category Menu, Bookmarks and verify url is being updated per settings.
2. Switch store view after directly accessing Product via Home/Search Page, Search Suggestions, Bookmarks, Mini Cart/Cart and verify url is being updated per settings.
3. Verify on updating invalid url suffix user is displayed with 404 error.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
